guides/source/getting_started.md

```diff
@@ -7,1150 +7,1551 @@ This guide covers getting up and running with Ruby on Rails.
-TIP: JavaScript asset compression requires you to
-have a JavaScript runtime available on your system, in the absence
-of a runtime you will see an `execjs` error during asset compression.
-Usually macOS and Windows come with a JavaScript runtime installed.
-`therubyrhino` is the recommended runtime for JRuby users and is added by
-default to the `Gemfile` in apps generated under JRuby. You can investigate
-all the supported runtimes at [ExecJS](https://github.com/rails/execjs#readme).
+To see your Rails application, open http://localhost:3000 in your browser. You will see the default Rails welcome page:
 
-This will start up Puma, a web server distributed with Rails by default. To see
-your application in action, open a browser window and navigate to
-<http://localhost:3000>. You should see the Rails default information page:
+![Rails welcome page](images/getting_started/rails_welcome.png)
 
-![Rails startup page screenshot](images/getting_started/rails_welcome.png)
+It works!
 
-When you want to stop the web server, hit Ctrl+C in the terminal window where
-it's running. In the development environment, Rails does not generally
-require you to restart the server; changes you make in files will be
-automatically picked up by the server.
+This page is the *smoke test* for a new Rails application, ensuring that everything is working behind the scenes to serve a page.
 
-The Rails startup page is the _smoke test_ for a new Rails
-application: it makes sure that you have your software configured correctly
-enough to serve a page.
+To stop the Rails server anytime, press `Ctrl-C` in your terminal.
 
-### Say "Hello", Rails
+### Automatic Reloading in Development
 
-To get Rails saying "Hello", you need to create at minimum a *route*, a
-*controller* with an *action*, and a *view*. A route maps a request to a
-controller action. A controller action performs the necessary work to handle the
-request, and prepares any data for the view. A view displays data in a desired
-format.
+Developer happiness is a cornerstone philosophy of Rails and one way of achieving this is with automatic code reloading in development.
 
-In terms of implementation: Routes are rules written in a Ruby [DSL
-(Domain-Specific Language)](https://en.wikipedia.org/wiki/Domain-specific_language).
-Controllers are Ruby classes, and their public methods are actions. And views
-are templates, usually written in a mixture of HTML and Ruby.
+Once you start the Rails server, new files or changes to existing files are detected and automatically loaded or reloaded as necessary. This allows you to focus on building without having to restart your Rails server after every change.
 
-Let's start by adding a route to our routes file, `config/routes.rb`, at the
-top of the `Rails.application.routes.draw` block:
+You may also notice that Rails applications rarely use `require` statements. Rails uses naming conventions to require files automatically so you can focus on writing your application code.
```

Comment:
what if your background is completely different, maybe you don't know how Ruby works either, therefore, spotting the difference may not be obvious.
I would keep this out, or move to the app folder structure and mention that Rails strongly recommends sticking to naming conventions to avoid `require`.

Reply:
Or: "You may also notice that Rails applications rarely use `require` statements like you may have seen in other programming languages." 
(I swear we had something like this before.)

Reply:
Added @AmandaPerino's line. I think we had that earlier too but it must have gotten lost somewhere.